e2e/install_mode_test.go

```diff
@@ -0,0 +1,132 @@
+package e2e_test
+
+import (
+	"strings"
+	"time"
+
+	. "github.com/maistra/istio-workspace/e2e/infra"
+	"github.com/maistra/istio-workspace/test"
+	"github.com/maistra/istio-workspace/test/shell"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator End to End Tests", func() {
+
+	Context("using install mode", func() {
+
+		var (
+			operatorNamepace string
+			namespaces       []string
+
+			cleanEnvVariable func()
+		)
+		BeforeEach(func() {
+			if RunsAgainstOpenshift {
+				Skip("Only run on microk8s cluster for complete isolation")
+			}
+
+			namespaces = []string{}
+			operatorNamepace = generateNamespaceName()
+			cleanEnvVariable = test.TemporaryEnvVars("OPERATOR_NAMESPACE", operatorNamepace)
+		})
+
+		AfterEach(func() {
+			cleanEnvVariable()
+			for _, namespace := range namespaces {
+				cleanupNamespace(namespace)
+			}
+			cleanupNamespace(operatorNamepace)
+		})
+		CreateNamespace := func() {
+			for _, namespace := range namespaces {
+				<-shell.Execute(NewProjectCmd(namespace)).Done()
+			}
+		}
+		WatchListExpression := func() string {
+			return strings.Join(namespaces, ",")
+		}
+		VerifyWatchList := func(validateNamespaces ...string) {
+			for _, watchNs := range validateNamespaces {
+				ikeCreate := RunIke(shell.GetProjectDir(), "create",
+					"--deployment", watchNs+"-v1",
+					"-n", watchNs,
+					"--route", "header:x-test-suite=smoke",
+					"--image", "x:x:x", // never used
```

Comment:
Maybe worth explaining a bit more in the comment why we have this "dummy" deployment at all :) We might forget at one point.

Reply:
https://github.com/maistra/istio-workspace/pull/680/commits/8945c58858762b4f17a366eb4fc938ce18b869fa